controller/controller_test.go

```diff
@@ -173,6 +173,36 @@ func (s *S) TestDeleteApp(c *C) {
 	}
 }
 
+func (s *S) TestDeleteAppUUIDName(c *C) {
+	for _, useName := range []bool{false, true} {
+		app := s.createTestApp(c, &ct.App{Name: random.UUID()})
+
+		var appID string
+		if useName {
+			appID = app.Name
+		} else {
+			appID = app.ID
+		}
+		c.Assert(s.c.DeleteApp(appID), IsNil)
+
+		_, err := s.c.GetApp(appID)
+		c.Assert(err, Equals, controller.ErrNotFound)
+	}
+}
+
+func (s *S) TestDeleteNonExistingApp(c *C) {
+	for _, useUuid := range []bool{false, true} {
+		var appID string
+
+		if useUuid {
+			appID = "foobar"
+		} else {
+			appID = random.UUID()
+		}
+		c.Assert(s.c.DeleteApp(appID), NotNil)
```

Comment:
This should assert that the error is `controller.ErrNotFound` (and modify the code to return that if it doesn't already).